internal/services/communication/helper/domain_record.go

```diff
@@ -0,0 +1,53 @@
+package helper
+
+import (
+	"github.com/hashicorp/go-azure-helpers/lang/pointer"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/communication/2023-03-31/domains"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+)
+
+type DomainVerificationRecords struct {
+	Type  string `tfschema:"type"`
+	Name  string `tfschema:"name"`
+	Value string `tfschema:"value"`
+	TTL   int    `tfschema:"ttl"`
```

Comment:
This should be an `int64`, use of `int` is being deprecated in resource and data source models.
```suggestion
	TTL   int64  `tfschema:"ttl"`
```